### PR TITLE
Separate DoubleClick RTC tests into their own file

### PR DESCRIPTION
**Pull request by bradfrizzell**
Also switch to using describes.realWin

This helps deal with state leakage, and makes the tests more maintainable. Part of a larger refactor effort for the A4A tests.

### CODE REVIEW

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 1]
```diff
@@ -661,11 +661,12 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
    *   the ad request url.
    * @private
    */
-  executeRtc_() {
+  executeRtc_(opt_doc) {
+    const doc = opt_doc || document;
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: this.element.ownerDocument instead of document

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+} from '../amp-ad-network-doubleclick-impl';
+import {createElementWithAttributes} from '../../../../src/dom';
+import {installDocService} from '../../../../src/service/ampdoc-impl';
+import {Xhr} from '../../../../src/service/xhr-impl';
+
+// Uncomment out the next line when testing locally
+// import {AmpAd} from '../../../amp-ad/0.1/amp-ad';
+
+function setupForAdTesting(fixture) {
+  installDocService(fixture.win, /* isSingleDoc */ true);
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js]: Do you still need this when using realWin?

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+  }
+
+  function setRtcConfig(rtcConfigJson) {
+    const rtcConf = env.win.document.getElementById('amp-rtc');
+    rtcConf.innerText = JSON.stringify(rtcConfigJson);
+  }
+
+  beforeEach(() => {
+    resetRtcStateForTesting();
+    return createIframePromise().then(fixture => {
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js]: Ditto... realWin shouldn't require wrapping in an iframe anymore and should load all of the services.  Instead you should refer to env.win & env.win.document

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js (hunk 2 above)
Comment (bradfrizzell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js]: no, removed.

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js (hunk 1 above)
Comment (bradfrizzell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: done

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js (hunk 3 above)
Comment (bradfrizzell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js]: done

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+describes.realWin('DoubleClick Fast Fetch RTC', {amp: true}, env => {
+  let impl;
+  let element;
+  let sandbox;
+  let xhrMock;
+
+  beforeEach(() => {
+    sandbox = env.sandbox;
+    env.win.AMP_MODE.test = true;
+    resetRtcStateForTesting();
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js]: Do you need this in both beforeEach and afterEach?

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+      'layout': 'fixed',
+    });
+    const rtcConf = createElementWithAttributes(
+        env.win.document, 'script',
+        {type: 'application/json', id: 'amp-rtc'});
+    rtcConf.innerHTML = JSON.stringify({
+      endpoint: 'https://example-publisher.com/rtc/',
+    });
+    env.win.document.head.appendChild(rtcConf);
+    xhrMock = sandbox.stub(Xhr.prototype, 'fetchJson');
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js]: Should we be using Taymon's new mock service?

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js (hunk 4 above)
Comment (bradfrizzell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js]: I like having it there because it assures that my test can't break due to someone else not calling the reset in their test. It doesn't hurt any performance or anything, and helps with dealing with leakage from other tests.

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js (hunk 5 above)
Comment (bradfrizzell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js]: Need changes to the FetchMock library, should be done as separate PR

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 6]
```diff
@@ -657,15 +657,17 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
    * model.
    * The targeting info from the RTC updates the targeting info on
    * this object within mergeRtc.
+   * @param {Document=} opt_doc Optional document for testing.
    * @return {?Promise<?Object>} An object of parameters to add to
```
Comment (bradfrizzell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: done
